datafusion/common/src/utils/array_utils.rs

```diff
@@ -0,0 +1,171 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! Array concatenation utilities
+
+use std::sync::Arc;
+
+use crate::cast::as_generic_list_array;
+use crate::error::_exec_datafusion_err;
+use crate::utils::list_ndims;
+use crate::Result;
+use arrow::array::{
+    Array, ArrayData, ArrayRef, GenericListArray, NullBufferBuilder, OffsetSizeTrait,
+};
+use arrow::buffer::OffsetBuffer;
+use arrow::datatypes::{DataType, Field};
+
+/// Concatenates arrays
+pub fn concat_arrays(args: &[ArrayRef]) -> Result<ArrayRef> {
```

Comment:
This was pulled into datafusion-common in order to be used across datafusion-functions and datafusion-functions-nested? I'd prefer to keep it in datafusion-functions, as datafusion-functions-nested already has a dependency on that crate.